### PR TITLE
Create `zk-explorer` app, implement project submission page

### LINKED ISSUE #1825: [FEAT] Implement Project/Circuits Upload Flow
## Description
Implement the upload flow for users to submit ZK projects and circuits to the registry via a GitHub link. This system must validate the presence of circuits within the repository and prevent duplicate entries. The system should also ensure that the user has an account before beginning the upload process. User can access '+New Project' from user's personal dashboard to initiate the upload process.

### User Flow and Features

**GitHub Repository Linking**

- Implement an input for users to submit the GitHub repository URL.
- Validate the URL to ensure it points to a repository containing `.circom` files and other relevant circuit file types (`.leo`, `.nr`, `.rs` etc.). Confirm the presence of circuits within the GitHub repository to proceed with the submission.
- Perform a check against the system to ensure the project is not already registered, preventing duplicate entries.

**Circuit Metadata Extraction**
- Upon successful repository linking, automatically generate a list of available circuits, displaying essential details like the circuit name, the path, and the last updated timestamp.
- Provide users with the option to enrich circuit data with additional descriptions, number of constraints, inputs, outputs, and other pertinent details.

**Project Information Display**
- Utilize GitHub's API to extract and present project information for user verification, including:
  - Project image preview
  - Project name and author
  - Licensing information
  - List of contributors
  - Audits, with an option to upload audit reports
  - Project Description (TBD): generate an initial description of the project that users can edit and refine (i.e., via chatgpt).

**Additional Project Details**

- Fields for users to input or edit further project details such as the official website, documentation link, social media handles, and tagging for categorization.

Upon submission, generate a project overview page with the data populated from the GitHub API and the user's additional inputs, providing a comprehensive view of the project and its circuits.

### Screenshot
Screenshot 2023-11-10 at 10 31 34 PM

### PR DESCRIPTION
**Pull request by yurixander**
## Summary of changes

_Provide a detailed description of proposed changes._

- Created and integrated the `zk-explorer` app into the monorepo, which is based in Next.js and utilizes a shared technology stack including Tailwind, React, etc. No new dependencies were added (#1822).
- Added `start:zk-explorer` and `build:zk-explorer` scripts to the monorepo's `package.json` file.
- Added `zk-explorer` to the `README` file along with a short description.
- Created basic `README` for `zk-explorer` project.
- Created responsive design implementation of the project submission page (#1825).
- Created `GitHubOAuthButton` component to be used for GitHub OAuth authentication flow.
- Created barebones `Header` component implementation, which includes the use of breadcrumbs, search, and GitHub OAuth sign in button.
- Added `NEXT_PUBLIC_GITHUB_CLIENT_ID` environment variable entry to `.env.example`, which helps specify the GitHub application's client ID to be used during the OAuth authentication flow.
- Added client-size GitHub project URL validation to the project submission input using Regex.

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [ ] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [x] `apps/zk-explorer`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)

- Closes #1825

### Screen Recording

_If possible provide a screen recording of proposed change._

https://github.com/webb-tools/webb-dapp/assets/101931215/ef219322-300a-4823-ace4-98833738de9c

---

### Code Checklist

_Please be sure to add .stories documentation if any additions are made to `libs/webb-ui-components`._

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/zk-explorer/public/favicon.ico
Comment (vutuanlinh2k2) [inline apps/zk-explorer/public/favicon.ico:1]: remove empty file

File: apps/zk-explorer/public/.gitkeep
Comment (vutuanlinh2k2) [inline apps/zk-explorer/public/.gitkeep:1]: remove empty file

File: apps/zk-explorer/app/page.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import { useState } from 'react';
+import { IconSize } from "../../../dist/libs/icons/types";
+import { Link } from '@webb-tools/webb-ui-components/components/Link';
+
+export default function Index() {
+  const [githubUrl, setGithubUrl] = useState('');
+  const [isValidGithubUrl, setIsValidGithubUrl] = useState(false);
+  const [isGithubUrlDisplayingError, setIsGithubUrlDisplayingError] = useState(false);
+
+  const validateGithubUrl = (url: string): boolean => {
```
Comment (vutuanlinh2k2) [inline apps/zk-explorer/app/page.tsx:14]: consider move this into utils folder or utils.ts file

File: apps/zk-explorer/components/GitHubOAuthButton.tsx
Comment (vutuanlinh2k2) [inline apps/zk-explorer/components/GitHubOAuthButton.tsx:1]: consider turn this into a folder and move all the types into types.ts file

File: apps/zk-explorer/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+'use client';
+
+import { Input, Dropdown, DropdownBasicButton, DropdownBody, MenuItem, Typography, Breadcrumbs, BreadcrumbsItem } from '@webb-tools/webb-ui-components';
+import { ThreeDotsVerticalIcon, Search, CheckboxBlankCircleLine } from '@webb-tools/icons';
+import { FC } from 'react';
+import { GitHubOAuthButton } from './GitHubOAuthButton';
+
+export const Header: FC<unknown> = () => {
+  const githubOAuthClientId = process.env.NEXT_PUBLIC_GITHUB_CLIENT_ID;
```
Comment (vutuanlinh2k2) [inline apps/zk-explorer/components/Header.tsx:9]: in next.config.js, follow this [code](https://github.com/webb-tools/webb-dapp/blob/develop/apps/hubble-stats/next.config.js#L15) to make NEXT_PUBLIC_GITHUB_CLIENT_ID accessible to the client

**Review by vutuanlinh2k2 — CHANGES_REQUESTED** [review]
Great work @yurixander ⭐ ⭐ ⭐ 

The CI is currently broken. Rember to rub `yarn lint` and `yarn format` before pushing to the remote branch

**Comment by yurixander** [comment]
Thanks for the feedback, @vutuanlinh2k2!

* I've organized the types, utility functions, and also removed the empty files as directed.
* Addressed the issue regarding the environment variable.
* I've setup a local Git pre-commit hook to depend on `yarn lint && yarn format:check` so that I don't forget again 😄

File: apps/zk-explorer/app/page.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    <main className="flex flex-col gap-6 pt-6">
+      <Typography variant="h4" fw="bold">
+        Upload Project
+      </Typography>
+
+      <div className="flex flex-col sm:flex-row gap-6 items-start">
+        <Card className="max-w-[780px]">
+          <Typography variant="h5">1. GitHub Repository URL:</Typography>
+
+          <Input
```
Comment (vutuanlinh2k2) [inline apps/zk-explorer/app/page.tsx:30]: Consider creating a new component for this Input and moving all the hooks into it. We should handle client actions in components rather making the whole page client-side

**Review by vutuanlinh2k2 — CHANGES_REQUESTED** [review]
Just one more thing that I think we should change, great work @yurixander

**Review by monaiuu — APPROVED** [review]
Looks good.

@AtelyPham can we setup netlify deployment preview for `zk-explorer`?

**Review by vutuanlinh2k2 — APPROVED** [review]

**Pull request merged**